File: Python/set-mismatch.py
class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        numsSum = sum(list(range(1, len(nums) + 1)))
        numsSet = set()
        res = list()

        for num in nums:
            if num in numsSet:
                res.append(num)
            else:
                numsSet.add(num)
                numsSum -= num
        res.append(numsSum)

        return res

File: Swift/set-mismatch.swift
class Solution {
    func findErrorNums(_ nums: [Int]) -> [Int] {
        var sum = Array(1...nums.count).reduce(0, { $0 + $1 })
        var set = Set<Int>()
        var res = [Int]()

        for num in nums {
            if set.contains(num) {
                res.append(num)
            } else {
                set.insert(num)
                sum -= num
            }
        }
        res.append(sum)

        return res
    }
}
